### ISSUE TITLE
wiping a software-defined asset is confusing

### ISSUE THREAD

**Issue by sryza** [dagster-io/dagster#5472]
I went to the asset catalog, selected a software-defined asset, and hit "Wipe asset". I expected it to disappear from the list. Instead, nothing happened.  I later realized it wouldn't make sense for it to disappear from the list, because its definition still exists.

A couple ideas:
* Rename "Wipe asset" to "Wipe materializations"
* Give some indication that the wipe was successful. E.g. with a toast or by adding a permanent column with the number of materializations that can go to 0.

**Comment by bengotow** [comment]
This seems like a good suggestion and a quick fix to me. The only caveat is that the button `Wipe assets` changes to `Wipe 1 asset` when you select one, and now it'd need to say `Wipe materializations for 1 asset`. I think that's a bit wordy so we can probably just say `Wipe materializations (1)`  and explain in the toast what just happened.

I think it'd be nice to have materialization count in there but that'd require a backend change and i'm not sure whether the query performance would be acceptable for a list level count subquery.

**Comment by bengotow** [comment]
Hmm @sryza what do you think of something like this in the confirmation dialog? Thinking it might also be good to explain that if you DO want a SDA to disappear, you also need to remove it's definition from code?

<img width="1021" alt="image" src="https://user-images.githubusercontent.com/1037212/140669646-780439a9-9536-477e-9a6f-622220320325.png">

**Comment by sryza** [comment]
Even those "Wipe materializations for 1 asset" is more verbose, I think "Wipe materializations (1)" is a little misleading - i.e. it's easy to interpret (1) as the number of materializations, not the number of assets.

For the message, we should say "Software-defined assets" instead of "Statically defined assets" and "repository" instead of "code".

**Comment by bengotow** [comment]
Ahh just saw this, will make another small diff to incorporate these changes
